packages/link/src/components/FloatingLink/FloatingLinkUrlInput.tsx

```diff
@@ -31,7 +34,13 @@ export const useFloatingLinkUrlInput = (
 ) => {
   const onChange: React.ChangeEventHandler<HTMLInputElement> =
     React.useCallback((e) => {
-      floatingLinkActions.url(e.target.value);
+      const value = e.target.value;
+
+      if (isEncoded(value)) {
+        floatingLinkActions.url(value);
+      } else {
+        floatingLinkActions.url(encodeURI(value));
+      }
```

Comment:
Thanks for the PR! Can you refactor this logic into a utility function `encodeUrlIfNeeded.ts` and add tests to make sure there are no edge cases?

```ts
const isEncoded = url !== decodeURIComponent(url);
return isEncoded ? url : encodeURI(url);
```